crates/cast/bin/cmd/rpc.rs

```diff
@@ -29,16 +31,29 @@ pub struct RpcArgs {
     #[arg(long, short = 'w')]
     raw: bool,
 
+    /// Timeout for the rpc request
+    ///
+    /// The timeout will be used to override the default timeout for RPC.
+    /// Default value is 45s
+    #[arg(long, env = "ETH_RPC_TIMEOUT")]
+    pub timeout: Option<u64>,
```

Comment:
I think we should add this to Config and RpcOpts, so that it's always available in get_provider across the codebase

Reply:
cc @mattsse

Reply:
I tried this as first option in previous commit and it was conflicting with --timeout for other commands.

Reply:
So it would be possible but the flag should be named differently

Reply:
The other commands should also use the config's and RpcOpts timeout

Reply:
Hi @DaniPopes  just following up on this so we can complete the feature.

Do you believe we should rename the flags of the other commands to something else which will allow to add this timeout to RpcOpts? Their functionality seem different to me so it doesn't seem wise to use the same `RpcOpts::timeout` for everything.

Reply:
Yes I think so, a global timeout flag/config makes sense
cc @mattsse @klkvr for more opinions

Reply:
yeah this should indeed be available on RpcOpts level/global level so we can reuse this everywhere

Reply:
Thanks for the feedback. What about the other `--timeout` flags that conflict with this one if I move it to RpcOpts?

Reply:
Would it be acceptable if I name it `--request_timeout`?

Reply:
let's use `--rpc-timeout` for this new one you add and keep `--timeout` for the other for backward compatibility reasons

Reply:
Refactored!